File: android/Estacionamento/app/src/main/java/com/example/android/estacionamento/view/MainActivity.java
package com.example.android.estacionamento.view;

import android.app.Activity;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.android.estacionamento.R;
import com.example.android.estacionamento.model.Carro;
import com.example.android.estacionamento.service.EstacionamentoAPI;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private final static EstacionamentoAPI api = EstacionamentoAPI.retrofit.create(EstacionamentoAPI.class);


    private ListView cars_listView;

    CarAdapter carAdapter;

    private TextView price_textView;

    private static final int PRICE_PARKING = 5;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        cars_listView = (ListView)findViewById(R.id.list_cars_view);

        final Activity activity = this;

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, EditorActivity.class);
                startActivity(intent);
            }
        });

        //Obter todos os carros
        getAllCars("");

        carAdapter = new CarAdapter(this, new ArrayList<Carro>());
        cars_listView.setAdapter(carAdapter);

        cars_listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                //se quiser pegar o evento de clique do carro atual descomente o codigo abaixo
                Carro carroAtual = carAdapter.getItem(position);
                exibirModal(carroAtual);
            }
        });
    }

    private void exibirModal(Carro carroAtual){
        AlertDialog.Builder builder;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            builder = new AlertDialog.Builder(this, android.R.style.Theme_Holo_Dialog_NoActionBar);
        } else {
            builder = new AlertDialog.Builder(this);
        }

        int totalHoursParking = carroAtual.getValorTotalEstacionamento() / PRICE_PARKING;

        builder.setTitle("Estacionamento")
                .setMessage("Seu carro está " + totalHoursParking + " hora(s) no estacionamento")
                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        //clicou no ok
                    }
                })
                .setNegativeButton(android.R.string.no, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        //clicou no cancelar
                    }
                })
                .setIcon(R.drawable.ic_time_to_leave_black_24dp)
                .show();
    }

    public void getAllCars(String placa){

        final Call<List<Carro>> carros = api.getAllCars(placa);

        cars_listView = (ListView)findViewById(R.id.list_cars_view);

        carros.enqueue(new Callback<List<Carro>>() {
            @Override
            public void onResponse(Call<List<Carro>> call, Response<List<Carro>> response) {
                List<Carro> carrosResponseBody = response.body();

                carAdapter.addAll(carrosResponseBody);
                cars_listView.setAdapter(carAdapter);

            }
            @Override
            public void onFailure(Call<List<Carro>> call, Throwable t) {
                Toast.makeText(MainActivity.this, "Não foi encontrado nenhum carro", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

}

File: android/Pokedex/app/src/main/java/dornel/com/pokedex/MainActivity.kt
package dornel.com.pokedex

import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.view.View.GONE
import android.view.View.VISIBLE
import com.google.gson.JsonObject
import dornel.com.pokedex.controller.Api
import dornel.com.pokedex.controller.Reqs
import dornel.com.pokedex.model.Pokemon
import dornel.com.pokedex.util.ComparePokemons
import dornel.com.pokedex.util.JsonConverter
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class MainActivity : AppCompatActivity() {

    private val pokemons: ArrayList<Pokemon> = ArrayList()
    private val context: Context = this

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recycler_view.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
        recycler_view.layoutManager = LinearLayoutManager(this)

        val adapter = PokemonAdapter(pokemons, context)
        recycler_view.adapter = adapter

        var total = 0
        do{
            total++
            progress_loader.visibility = GONE
            Reqs.getPokemon(total).enqueue(object : Callback<JsonObject> {
                override fun onFailure(call: Call<JsonObject>?, t: Throwable?) {
                }

                override fun onResponse(call: Call<JsonObject>?, response: Response<JsonObject>?) {
                    val poke: Pokemon = JsonConverter.convertJsonToPoke(response?.body())
                    pokemons.add(poke)
                    pokemons.sortBy { pokeItem -> pokeItem.id }
                    adapter.notifyDataSetChanged()
                }
            })
        }while (10 >= total)
    }

    private fun registerPokemon(){
        val intent = Intent(this, RegisterActivity::class.java)
        startActivity(intent)
    }


}
